### ISSUE TITLE
Dynamo 0.9 for Revit crashing and producing lots of nulls

### ISSUE THREAD

**Issue by jluebker** [DynamoDS/Dynamo#5850]
Lately my dynamo for Revit has been crashing a lot (even though I am running the latest stable build)... Sandbox works just fine though.

I was also getting a whole lot of nulls for no reason  in this graph (even though the nodes are producing geometry --- or at least some of them are. It runs just fine in the sandbox though. 

![image](https://cloud.githubusercontent.com/assets/4041748/11847099/97209584-a3d0-11e5-87a1-7cdd48e4725c.png)

Here are the details of the latest crash:

The input argument "materialId" of function `anonymous-namespace'::TessellatedFace_constructor or one item in the collection is null at line 96 of file d:\sundial\2015_px64\source\api\revitapi\gensrc\APITessellatedShapeBuilder.cpp.
Parameter name: materialId

   at APIElementIdTypeConverter.ManagedToNative(ElementId managedObj, ElementId\* nativeObj, ManagedToNativeParameter convParam)
   at ?A0x91c5a93f.TessellatedFace_constructor(IList`1 outerLoopVertices, ElementId materialId)
   at Autodesk.Revit.DB.TessellatedFace..ctor(IList`1 outerLoopVertices, ElementId materialId)
   at Revit.GeometryConversion.ProtoToRevitMesh.ToRevitType(Solid solid, TessellatedShapeBuilderTarget target, TessellatedShapeBuilderFallback fallback, ElementId MaterialId, Boolean performHostUnitConversion)
   at Dynamo.Applications.ViewModel.RevitWatch3DViewModel.RevitGeometryObjectFromGraphicItem(IGraphicItem item, List`1& geoms)
   at Dynamo.Applications.ViewModel.RevitWatch3DViewModel.Draw()
   at Dynamo.Applications.ViewModel.RevitWatch3DViewModel.OnNodePropertyChanged(Object sender, PropertyChangedEventArgs e)
   at System.ComponentModel.PropertyChangedEventHandler.Invoke(Object sender, PropertyChangedEventArgs e)
   at Dynamo.Core.NotificationObject.RaisePropertyChanged(String propertyName)
   at Dynamo.Models.NodeModel.set_IsVisible(Boolean value)
   at Dynamo.Models.NodeModel.UpdateValueCore(UpdateValueParams updateValueParams)
   at Dynamo.Models.WorkspaceModel.UpdateModelValue(IEnumerable`1 modelGuids, String propertyName, String value)
   at Dynamo.Models.DynamoModel.UpdateModelValueImpl(UpdateModelValueCommand command)
   at Dynamo.Models.DynamoModel.ExecuteCommand(RecordableCommand command)
   at Dynamo.ViewModels.WorkspaceViewModel.ShowHideAllGeometryPreview(Object parameter)
   at MS.Internal.Commands.CommandHelpers.CriticalExecuteCommandSource(ICommandSource commandSource, Boolean userInitiated)
   at System.Windows.Controls.MenuItem.InvokeClickAfterRender(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by jluebker** [comment]
Update: the nulls are fixed when I turn off the preview in revit and the number of crashes reduces. But it is still crashing more than it should be or than it does in the sandbox.

**Comment by kronz** [comment]
@jluebker I suspect you have a really old version of Rhynamo.  Looks like this is the version that was made with Python nodes inside of Custom nodes.  Try updating to the latest version of that package?  It's pretty different now.

**Comment by monikaprabhu** [comment]
@jluebker are you still be able to reproduce this? Also are you using English revit+Dynamo or localized. We have seen crash related to preview in Revit in localized sessions. fix is here. https://github.com/DynamoDS/DynamoRevit/pull/795. 
I would like to know if your crash is related. So please test on the latest daily build and let me know.

**Comment by jluebker** [comment]
I am having crashes but I’m pretty sure they are related to flux. Though I did have a series of crash not related to flux when I submitted this. Your description however leads me to believe that it was the spring nodes familyinstance.bygeoemtry that I was working with. I was creating families using freeformelments and was neglecting to apply a material, which might cause the crash you described.

Jeremy Luebker
Arch II
t 415.546.2957
c 510.881.3682
Perkins+Will
